lms/djangoapps/discussion_api/api.py

```diff
@@ -697,6 +696,30 @@ def update_comment(request, comment_id, update_data):
     return api_comment
 
 
+def get_thread(request, thread_id):
+    """
+    Retrieve a thread.
+
+    Arguments:
+
+        request: The django request object used for build_absolute_uri and
+          determining the requesting user.
+
+        thread_id: The id for the thread to retrieve
+
+    Raises:
+
+        PermissionDenied: if user does not have permission to view thread
+
+    """
+    cc_thread, context = _get_thread_and_context(request, thread_id)
+    if has_permission(cc_thread, context):
```

Comment:
Could we add a test for this as well? That someone else enrolled in the course, but not the author or staff, is able to fetch the thread.